assignments/array-methods.js

```diff
@@ -55,29 +55,67 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+
 let fullName = [];
-console.log(fullName);
+runners.forEach(list => {
+	fullName.push(`${list.first_name} ${list.last_name}`)
+})
+console.log(fullName)
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+// let allCaps = [];
+let allCaps = runners.map(list => list.first_name.toUpperCase())
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// let largeShirts = [];
+let largeShirts = runners.filter(list => list.shirt_size === "L")
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+// let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((total, list) => total + list.donation, 0)
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
-
+// Need a list of Names (only) that have donated at 100 Dollars
+let over100 = runners.filter(list => list.donation > 100)
+	over100.forEach((list, index) => console.log(`${over100[index].first_name} ${over100[index].last_name}`))
+	
 // Problem 2
+//A list of people with gmail email
+let yahooEmail = runners.filter(list => list.email.includes("yahoo"))
+console.log(yahooEmail);
+
+// Problem 3
+// an array containing each unique company name, and how many times each company is reused
+let companies = runners.map(list => list.company_name)
+// console.log(companies);
+companies.sort();
+
+let uniqueCompanies = [];
+let count = 1; 
+
+companies.forEach((list, i, arrComp) => {
```

Comment:
I know we went over this with maks, but your code is very close to what he demonstrated. it will work properly with just a few small tweaks.

```
companies.forEach((company, i, array) => {
  if (company === array[i+1]) ++count;
  else {
    uniqueCompanies.push({[company]: count});
    count = 1;
  }
});
```